tools/python/boutiques/searcher.py

```diff
@@ -88,6 +89,13 @@ def create_results_list_verbose(self, results):
                                       ("SCHEMA VERSION", schema_version),
                                       ("CONTAINER", container),
                                       ("TAGS", other_tags)])
+            for k, v in list(result_dict.items()):
+                if sys.version_info[0] < 3:
+                    if isinstance(v, unicode):
+                        result_dict[k] = v.encode('ascii', 'xmlcharrefreplace')
+                elif isinstance(v, str):
+                    result_dict[k] = \
+                        v.encode('ascii', 'xmlcharrefreplace').decode()
```

Comment:
Could you add some of the comments you made in the PR text here? That would be useful in the future.

I understand that this code replaces UTF-8 characters by XML char refs, but I am not sure that it is suitable. `bosh` outputs aren't supposed to be printed only in XML documents, but also in terminals and other UTF-8-enabled devices. Perhaps more info on #386 would be useful to understand why this is a good fix.

Reply:
Also, we would need such a fix to be tested.

Reply:
Minimal code to see this error:
```
print( u"\xef")
```

It works on the terminal in a mac because the terminal has utf-8 support. It works in python2.7 on a mac, as well. However, when SSHing into a terminal, it will break for me because my terminal is not willing to print utf-8 to standard out. This appears not to even be about supporting a version of python, but supporting different terminals (with different encodings). On my terminal I can try to force utf-8 encodings and then it prints junk characters (more than one, even).

The proposed solution above prints the xmlcharrefreplace \&#239; (which if I didn't escape would re-encode in github as the expected '&#239;'). It works consistently on all systems, but is of course not pretty. 

For searcher, the output generation is all isolated here. In theory, I could do something that detected whether the terminal supported utf-8. For example, on a mac:
```
import sys
print(sys.stdout.encoding)
```

will show you that you have 'UTF-8' and so I can leave things as-is. My terminal gives the encoding 'ANSI_X3.4-1968' so I can't do anything fun and I need the above ugly (but consistent) solution.

So, uniform ugly, or I wrap the re-formatting for-loop in a check so that it is only ugly when UTF-8 isn't supported? I guess Travis supports utf-8, which is why you don't see this in CI?